library/std/src/os/windows/ffi.rs

```diff
@@ -124,11 +126,95 @@ pub trait OsStrExt: Sealed {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     fn encode_wide(&self) -> EncodeWide<'_>;
+
+    /// Compares two `OsStr` by using the Windows system implementation.
+    /// This performs a case-insensitive comparison of UTF-16 code units using the system case mappings.
+    /// The comparison is locale-independent, but exact results may depend on the Windows version,
+    /// file system, and system settings.
+    ///
+    /// This is the correct way to compare various strings on Windows:
+    /// environment variable keys, registry keys, file paths and resource handle names are all case-insensitive.
```

Comment:
As per https://github.com/rust-lang/rust/issues/86007#issuecomment-855013160 this might still not be the correct way for dealing with file paths. What should we say about this in the documentation?

Reply:
Personally, I'm uncertain. Perhaps remove "file paths" from that list but then add a note that file paths may have different rules? I think `system_cmp` could still perhaps be useful as a heuristic (i.e. these two paths may be equal) so long as it isn't relied upon for file paths.

But I don't think there's any OS support for comparing file paths other than opening both files and seeing if they point to the same file. Which obviously can't be done if one of the paths doesn't yet exist (or they can't be opened for some other reason). I mean, `system_cmp` may work for file paths most of the time but when it doesn't it could be a footgun. Or at least surprising when two "equal" paths point to different files.

Reply:
Yeah comparing file paths is more complex, also due to the possibility of relative paths, verbatim paths, UNC prefixes, symlinks, etc; there are many ways in which two different paths can be equivalent. 

It seems more appropriate to find a separate correct solution for that, so I'll leave it out of the documentation in this PR.